### ISSUE TITLE
Fill horizontal space

### ISSUE THREAD

**Issue by xeruf** [hzeller/timg#66]
I want to give timg multiple images scaled to a certain height, and want them to be placed right next to each other without worrying about columns.

**Comment by hzeller** [comment]
If you just want to fill the available space in the width and use whatever makes sense in the height you can tell it to have the number of columns you need, say you want to have 5 columns:

``` 
timg --grid=5x1 *.jpg
```

The 5x1 means: fill the width with 5 columns and take whatever space is needed up to the height of the terminal.

If you want to limit the height in a particular way, you need to also give the geometry.

```
timg --grid=5x1 -g 100x50 *.jpg
```

Will make the images at most 50 characters high and attempts to fit 5 columns in the 100 character width.

Ideally, you'd just give the height and let `timg` figure out the width, but currently `timg` only does either full automatic or full manual (so here, we ideally would like to have something like `-gx50`, meaning only the height is specified). I think this would be a good feature to have; let me think about that.

**Comment by hzeller** [comment]
Alright, added the feature to only partially specify the geometry. So now, it is possible to tell `timg` to have a specific number of columns evenly spaced into the available width of the terminal whlie not exceeding a certain height:

```
timg --grid=5x1 -gx50`

**Comment by hzeller** [comment]
Does this solve your feature need ?

**Comment by xeruf** [comment]
No, unfortunately not, though that is a nice step - I don't want them evenly spaced but right next to each other, so I don't want a fixed grid at all.

**Comment by hzeller** [comment]
So you'd like to have something like a 'flowing' layout, in which we fill a row with as many images as possible of a particular height, and then start the next row. Depending on the width of the pictures we might get more or less in one row.
Something like that ?

**Comment by xeruf** [comment]
Exactly :)

**Comment by acxz** [comment]
I think this could be done by having an option to update the `x_offset` based on the image width instead of the fixed column size.

https://github.com/hzeller/timg/blob/cfd5f7a99401074ccfad31fa6234e94e5175074f/src/renderer.cc#L96

Essentially keep a running tally of the current x offset and as you loop through the images keep on adding the image width to the x_offset.
